conda/recipes/custreamz/meta.yaml

```diff
@@ -39,7 +39,7 @@ requirements:
     - python
     - rapids-build-backend >=0.3.0,<0.4.0.dev0
     - setuptools
-    - python-confluent-kafka >=1.9.0,<1.10.0a0
+    - python-confluent-kafka >=2.5.0,<2.6.0a0
```

Comment:
There aren't any `python-confluent-kafka` releases in the 1.x series that have Python 3.12 packages on conda-forge (https://anaconda.org/conda-forge/python-confluent-kafka).

Resulting in build errors like this:

```text
  LibMambaUnsatisfiableError: Encountered problems while solving:
    - package python-confluent-kafka-1.9.0-py310h5764c6d_0 requires python_abi 3.10.* *_cp310, but none of the providers can be installed
  
  Could not solve for environment specs
  The following packages are incompatible
  ├─ python-confluent-kafka >=1.9.0,<1.10.0a0  is installable with the potential options
  │  ├─ python-confluent-kafka [1.9.0|1.9.2] would require
  │  │  └─ python_abi 3.10.* *_cp310, which can be installed;
  │  ├─ python-confluent-kafka [1.9.0|1.9.2] would require
  │  │  └─ python_abi 3.7.* *_cp37m, which can be installed;
  │  ├─ python-confluent-kafka [1.9.0|1.9.2] would require
  │  │  └─ python_abi 3.8.* *_cp38, which can be installed;
  │  ├─ python-confluent-kafka [1.9.0|1.9.2] would require
  │  │  └─ python_abi 3.9.* *_cp39, which can be installed;
  │  └─ python-confluent-kafka 1.9.2 would require
  │     └─ python_abi 3.11.* *_cp311, which can be installed;
  └─ python 3.12**  is not installable because there are no viable options
     ├─ python [3.12.0|3.12.1|...|3.12.5] would require
     │  └─ python_abi 3.12.* *_cp312, which conflicts with any installable versions previously reported;
     └─ python 3.12.0rc3 would require
        └─ _python_rc, which does not exist (perhaps a missing channel).
```

([build link](https://github.com/rapidsai/cudf/actions/runs/10711254837/job/29737800673?pr=16745))

Pushed a commit (https://github.com/rapidsai/cudf/pull/16745/commits/8c206d77a73aa2251b68088f02c1fa288dcd1f86) which updates `python-confluent-kafka` and `librdkafka` to the latest minor version (the 2.5.x series) and continues the practice to pinning build and runtime dependencies within a single minor version of Kafka libraries.

Reply:
Updating the Kafka dependencies is fine with me.

cc: @jdye64 for awareness.